lib/rom/sql/extensions/postgres/types/range.rb

```diff
@@ -0,0 +1,223 @@
+          PAREN_LEFT  = '('.freeze
+          PAREN_RIGHT = ')'.freeze
+
+          def initialize(lower, upper, bounds = :'[)')
+            super
+          end
+
+          def exclude_begin?
+            bounds[0] == PAREN_LEFT
+          end
+
+          def exclude_end?
+            bounds[1] == PAREN_RIGHT
+          end
+        end
+      end
+
+      # @api public
+      module Types
+        # The list of range types supported by PostgreSQL
+        # @see https://www.postgresql.org/docs/current/static/rangetypes.html
+
+        @range_parsers = {
+          int4range: Sequel::Postgres::PGRange::Parser.new(
+            'int4range', SQL::Types::Coercible::Int
+          ),
+          int8range: Sequel::Postgres::PGRange::Parser.new(
+            'int8range', SQL::Types::Coercible::Int
+          ),
+          numrange:  Sequel::Postgres::PGRange::Parser.new(
+            'numrange', SQL::Types::Coercible::Int
+          ),
+          tsrange:   Sequel::Postgres::PGRange::Parser.new(
+            'tsrange', SQL::Types::Form::Time
+          ),
+          tstzrange: Sequel::Postgres::PGRange::Parser.new(
+            'tstzrange', SQL::Types::Form::Time
+          ),
+          daterange: Sequel::Postgres::PGRange::Parser.new(
+            'daterange', SQL::Types::Form::Date
+          )
+        }.freeze
+
+        # @api private
+        def self.range_read_type(name)
+          SQL::Types.Constructor(Values::Range) do |value|
+            pg_range =
+              if value.is_a?(Sequel::Postgres::PGRange)
+                value
+              elsif value && value.respond_to?(:to_s)
+                @range_parsers[name].(value.to_s)
+              else
+                value
+              end
+
+            Values::Range.new(
+              pg_range.begin,
+              pg_range.end,
+              [pg_range.exclude_begin? ? :'(' : :'[',
+               pg_range.exclude_end? ? :')' : :']']
+              .join('').to_sym
+            )
+          end
+        end
+
+        # @api private
+        def self.range(name, read_type)
+          Type(name) do
+            type = SQL::Types.Definition(Values::Range).constructor do |range|
+              format('%s%s,%s%s',
+                     range.exclude_begin? ? :'(' : :'[',
+                     range.lower,
+                     range.upper,
+                     range.exclude_end? ? :')' : :']')
+            end
+
+            type.meta(read: read_type)
+          end
+        end
+
+        Int4Range = range('int4range', range_read_type(:int4range))
+        Int8Range = range('int8range', range_read_type(:int8range))
+        NumRange  = range('numrange',  range_read_type(:numrange))
+        TsRange   = range('tsrange',   range_read_type(:tsrange))
+        TsTzRange = range('tstzrange', range_read_type(:tstzrange))
+        DateRange = range('daterange', range_read_type(:daterange))
+
+        module RangeOperators
+          def contains(_type, expr, value)
```

Comment:
I think we should rename it to `contain`, it’s a semi-consistent ruby’s convention, as in `respond_to?`, `start_with?`, `include?` etc follow this.